src/platform/packages/shared/kbn-opentelemetry-utils/src/from_traceparent.ts

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+import { SpanContext, TraceFlags } from '@opentelemetry/api';
+
+// W3C Trace Context header regex (case-insensitive)
+// version 2 hex, trace-id 32 hex, span-id 16 hex, flags 2 hex
+const TRACEPARENT_REGEX = /^[0-9a-f]{2}-[0-9a-f]{32}-[0-9a-f]{16}-[0-9a-f]{2}$/i;
+
+// Helpers to check if ID is not all zeros (spec prohibits all-zero ids)
+const isAllZeros = (str: string) => /^0+$/.test(str);
+
+export function fromTraceparent(traceparent: string): SpanContext | undefined {
```

Comment:
😮 I'm surprised that this helper is not provided in the `@opentelemetry` libraries

Reply:
me too, I'll have another look to see if I haven't missed anything.